### PR TITLE
Extract project code validation

### PR DESCRIPTION
**Pull request by KochTobi**
This extracts the validation logic for project codes

### CODE REVIEW

File: src/main/groovy/life/qbic/datamodel/dtos/projectmanagement/ProjectCodeValidator.groovy [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+/**
+ * <b>Project Code Validator</b>
+ *
+ * <p>Consumes a project code and throws a {@link ValidationException} in case it is not valid.</p>
+ *
+ * @since 2.12.0
+ */
+class ProjectCodeValidator implements Consumer<String> {
+    private static final Predicate<String> IS_VALID_PROJECT_CODE = Pattern.compile("Q[A-X0-9]{4}").asPredicate()
```
Comment (sven1103) [inline src/main/groovy/life/qbic/datamodel/dtos/projectmanagement/ProjectCodeValidator.groovy:17]: Small extensions, to match really single project codes, and not sample codes, entities or even text with a project code contained. A project code starts with a `Q` and ends after four additional alpha-numeric letters.

```suggestion
    private static final Predicate<String> IS_VALID_PROJECT_CODE = Pattern.compile("^Q[A-X0-9]{4}$").asPredicate()
```

**Review by sven1103 — CHANGES_REQUESTED** [review]
I suggest to move the validation class now out of the DTO package.

+1 for lambda support via the functional interfaces.

**Comment by sven1103** [comment]
Ah and the CL needs an update as well :)

File: src/main/groovy/life/qbic/datamodel/dtos/projectmanagement/ProjectCodeValidator.groovy (hunk 1 above)
Comment (KochTobi) [reply src/main/groovy/life/qbic/datamodel/dtos/projectmanagement/ProjectCodeValidator.groovy]: Nice catch 
image

**Review by sven1103 — APPROVED** [review]

**Pull request merged**